oak-run/src/main/java/org/apache/jackrabbit/oak/run/CreateGarbageCommand.java

```diff
@@ -0,0 +1,568 @@
+                    .accepts("orphansDepth", "the depth in the tree at which to create garbage nodes. Only applies to ORPHANS fullGC modes").withRequiredArg()
+                    .ofType(Integer.class).defaultsTo(1);
+            orphansLevelGap = parser
+                    .accepts("orphansLevelGap", "the gap in the tree between the first leaf gap orphan garbage node and its' parent. Only applies to ORPHANS fullGC modes").withRequiredArg()
+                    .ofType(Integer.class).defaultsTo(0);
+            numberOfRuns = parser
+                    .accepts("numberOfRuns", "the number of garbage generation runs to do. Only applies if greater than 1, " +
+                            "otherwise a single run will be done.").withRequiredArg()
+                    .ofType(Integer.class).defaultsTo(1);
+            generateIntervalSeconds = parser
+                    .accepts("generateIntervalSeconds", "the interval at which to generate a complete garbage count from createGarbageNotesCount. " +
+                            "Applies only if numberOfRuns is greater than 1.").withRequiredArg()
+                    .ofType(Integer.class).defaultsTo(60);
+        }
+
+        public CreateGarbageOptions parse(String[] args) {
+            super.parse(args);
+            return this;
+        }
+
+        String getSubCmd() {
+            List<String> args = getOtherArgs();
+            if (args.size() > 0) {
+                return args.get(0);
+            }
+            return CMD_CREATE;
+        }
+
+        public int getCreateGarbageNodesCount() {
+            return createGarbageNodesCount.value(options);
+        }
+
+        public int getGarbageNodesParentCount() {
+            return garbageNodesParentCount.value(options);
+        }
+
+        public int getGarbageType() {
+            return garbageType.value(options);
+        }
+
+        public int getOrphansDepth() { return orphansDepth.value(options);}
+
+        public int getOrphansLevelGap() { return orphansLevelGap.value(options); }
+
+        public int getNumberOfRuns() {
+            return numberOfRuns.value(options);
+        }
+
+        public int getGenerateIntervalSeconds() {
+            return generateIntervalSeconds.value(options);
+        }
+    }
+
+    @Override
+    public void execute(String... args) throws Exception {
+
+        try (Closer closer = Closer.create()) {
+            // Register the ExecutorService with Closer
+            closer.register(() -> {
+                // Properly shutdown the executor service
+                continuousRunExecutor.shutdown();
+                try {
+                    if (!continuousRunExecutor.awaitTermination(5, TimeUnit.SECONDS)) {
+                        continuousRunExecutor.shutdownNow();
+                    }
+                } catch (InterruptedException e) {
+                    throw new RuntimeException(e);
+                }
+            });
+
+            execute(closer, args);
+
```

Comment:
when used in continuous mode (eg by setting `--numberOfRuns 42`) then the code here terminates early and closes the Closer - preventing the continuous mode from practically ever running

Reply:
Fixed, thank you for the comment.